### PR TITLE
Make path_size_megabytes() more robust.

### PR DESCRIPTION
**Pull request by concretevitamin**
Saw some parsing errors that threw ValueError.

This change
- uses regex to look for 'total size is ...'
- on error, return -1 as estimated size, as it's used for warnings only

Tested (run the relevant ones):

- [x] Any manual or new tests for this PR (please specify below)
  - `sky launch test.yaml` with `workdir: .`; checked that sizes are correct
- [ ] All smoke tests: `pytest tests/test_smoke.py` 
- [ ] Relevant individual smoke tests: `pytest tests/test_smoke.py::test_fill_in_the_name` 
- [ ] Backward compatibility tests: `bash tests/backward_comaptibility_tests.sh`

### CODE REVIEW

File: sky/backends/backend_utils.py [hunk 1]
```diff
@@ -276,21 +276,36 @@ def _optimize_file_mounts(yaml_path: str) -> None:
...
+    rsync_output = ''
+    try:
+        rsync_output = str(
+            subprocess.check_output(
+                f'rsync {command_runner.RSYNC_DISPLAY_OPTION} '
+                f'{command_runner.RSYNC_FILTER_OPTION} '
+                f'{git_exclude_filter} --dry-run {path!r}',
+                shell=True))
+    except subprocess.CalledProcessError:
+        return -1
```
Comment (Michaelvll) [inline sky/backends/backend_utils.py:300]: should we output the stderr to `logger.warning` for better future debugging?

File: sky/backends/backend_utils.py [hunk 2]
```diff
@@ -276,21 +276,36 @@ def _optimize_file_mounts(yaml_path: str) -> None:
...
+    match = re.search(r'total size is (\d+)', rsync_output)
```
Comment (Michaelvll) [inline sky/backends/backend_utils.py:301]: Question: It seems there will be `,` separating the numbers, will `\d+` recognize it?
```
total size is 250,957,728  speedup is 330.19 (DRY RUN)
```

File: sky/backends/backend_utils.py [hunk 3]
```diff
@@ -276,21 +276,36 @@ def _optimize_file_mounts(yaml_path: str) -> None:
...
+    if match is not None:
+        try:
+            total_bytes = int(float(match.group(1)))
+            return total_bytes // (1024**2)
+        except ValueError:
+            pass  # Maybe different rsync versions have different output.
```
Comment (Michaelvll) [inline sky/backends/backend_utils.py:307]: Similarly, should we `logger.debug` this?

**Review by Michaelvll — APPROVED** [review]
Thanks for the changes @concretevitamin! LGTM. Some questions

File: sky/backends/backend_utils.py (hunk 2 above)
Comment (concretevitamin) [reply sky/backends/backend_utils.py:301]: My rsync:
```
» rsync --version                          
rsync  version 2.6.9  protocol version 29
```
What's your version?

Comment (Michaelvll) [reply sky/backends/backend_utils.py:301]: Mine is 
```
rsync --version                                                                                      
rsync  version 3.2.3  protocol version 31
```

I think I upgraded the rsync with homebrew once.

File: sky/backends/backend_utils.py (hunk 1 above)
Comment (concretevitamin) [reply sky/backends/backend_utils.py:300]: Added logger.debug(). Wdyt?

File: sky/backends/backend_utils.py (hunk 2 above)
Comment (concretevitamin) [reply sky/backends/backend_utils.py]: Good catch! Changed the pattern.

I did not proceed to test more versions as this is used for warning only & can fix on-demand.

**Review by Michaelvll — APPROVED** [review]

**Pull request merged**